### ISSUE TITLE
Clarifying the new AEAD robustness solution

### ISSUE THREAD

**Issue by samuel-lucas6** [jedisct1/libsodium-doc#131]
In the [AEAD Robustness](https://doc.libsodium.org/secret-key_cryptography/aead#robustness) section, the interactive protocol explanation is a lot better now, but the new non-interactive protocol solution isn't very clear. In [this](https://github.com/jedisct1/libsodium-doc/commit/5dfde30e2b2dd11c1e4314425c82f260948bbe92) commit you wrote ```H(k, nonce || ciphertext_tag)``` and mentioned using HMAC. However, I saw that you then [changed](https://github.com/jedisct1/libsodium-doc/commit/439aa4610b869e9150a9c7db264bb187092d8bef) the example to ```H(key || nonce || ciphertext_tag)``` and now mention BLAKE2. This raises two questions:

1. Is this meant to be a MAC? I presume the answer is yes considering that HMAC is still listed, but that format suggests that it's just an unkeyed hash.
2. If so, where does the key for the MAC come from? Unless the suggestion is to use the same key for the MAC and the AEAD, but then why would the ```key``` have to be concatenated with the ```nonce``` and ```ciphertext_tag``` as well? Plus implementing it in that way would probably get some criticism due to the key reuse.

I believe it would be beneficial to answer these questions in the documentation. There's a lot of confusion about this topic, and that's not surprising when it seems to be underdocumented and not explained in layman's terms anywhere, which seems to be a common problem in the field of cryptography. This page has to be the best summary currently available, but there's still some room for improvement.

Finally, just out of curiosity, what was the reasoning behind [removing](https://github.com/jedisct1/libsodium-doc/commit/3c20a03561cbfb6ab2443f1d1084c2e18f78cfff) the padding fix suggestion? Was it due to potential timing attacks? It seems to be the most discussed fix in papers on the topic, and it's also easy to implement.

**Comment by jedisct1** [comment]
Sorry, I don't do free support.
